bindings/pyroot/pythonizations/python/ROOT/_pythonization/_ttree.py

```diff
@@ -160,46 +201,10 @@ def _SetBranchAddress(self, bname, addr, *args, **kwargs):
     ```
     """
 
-    import cppyy
-    import ctypes
-
-    # Pythonization is only active for cppyy proxies (of type CPPInstance)
-    if isinstance(addr, cppyy._backend.CPPInstance):
-
-        addr_of_object = cppyy.addressof(instance=addr, byref=False)
+    branch = self.GetBranch(bname)
+    addr = _pythonize_branch_addr(branch, addr)
 
-        is_leaf_list = self.GetBranch(bname).IsA() is cppyy.gbl.TBranch.Class()
-
-        if is_leaf_list:
-            # If the branch is a leaf list, SetBranchAddress expects directory
-            # the address of the object that has the corresponding data
-            # members.
-            addr = addr_of_object
-        else:
-            # Otherwise, SetBranchAddress is expecting a pointer to the address
-            # of the object, and the pointer needs to stay alive. Therefore, we
-            # create a container for the pointer and cache it in the cppyy
-            # proxy.
-            addr_view = cppyy.gbl.array["std::intptr_t", 1]([addr_of_object])
-
-            if not hasattr(addr, "_set_branch_cached_pointers"):
-                addr._set_branch_cached_pointers = []
-            addr._set_branch_cached_pointers.append(addr_view)
-
-            addr = cppyy.addressof(instance=addr_view, byref=False)
-    elif hasattr(addr, "__array_interface__"):
-        # For NumPy arrays
-        addr = addr.__array_interface__["data"][0]
-    elif hasattr(addr, "buffer_info"):
-        # For the builtin array library
-        addr = addr.buffer_info()[0]
-
-    if not isinstance(addr, int):
-        raise TypeError(
-            "TTree::SetBranchAddress(): Unsupported type! Only PyROOT objects, arrays and memory addresses of type `int` are supported."
-        )
```

Comment:
We lost this check in the latest changes, is it not important?

Reply:
I realized it was a change of behavior from the existing code: the old code did a pass-through when it didn't know what to do with the address. Maybe it's better to keep the existing behavior, e.g. for the case where you already have a `c_void_p` or cppyy `void*` low-level view in hand.